util/image-upload.js

```diff
@@ -1,9 +1,20 @@
 import { By, until } from "selenium-webdriver";
 import { defaultWait } from "../config.js";
 import path from "path";
+import { ExtendedBy } from "./selection.js";
 
-export async function uploadImage(driver, element_id, imagePath) {
-    const imageField = await driver.findElement(By.id(element_id));
+/**
+ * @param {any} driver 
+ * @param {string} element_id 
+ * @param {number} nthElement 
+ * @param {string} imagePath 
+ */
+export async function uploadImage(driver, element_id, nthElement, imagePath) {
+    let imageField;
+    if(nthElement === 0)
+        imageField = await driver.findElement(By.xpath(`(//*[starts-with(@id, '${element_id}')])`));
+    else
+        imageField = await driver.findElement(By.xpath(`(//*[starts-with(@id, '${element_id}')])[${nthElement}]`));
```

Comment:
How can there be multiple fields with the same ID? Generally finding elements with an index is rather unstable and I think we should avoid it wherever possible. I admit though that I also used it sometimes

Reply:
I would love to know as well :sweat_smile: I mostly tried getting the tests to pass without the need for changes in the ui code.
I found some components that simply used an id on elements but that were also repeated in a list, which of course fundamentally breaks the idea of an id.
This does need refactoring, but I'd suggest to start that on the ui side first. What do you think?

Reply:
I just revisited this and realized thats not what was happening here. Even though there are some components that share the same id.

Consider the image upload feature for a badge.
There are two options to choose from, both input ids starting with "image_field".
This method now easily allows to choose the second one by passing in n = 1.

The problem arised from one of the badges where the image fields weren't numbered in ascending order from 0 but from 1.
This makes it easier and more stable if these indices ever change again, since we are still selecting the second occurrence in the DOM rather than the exact id.